terraform-modules/cloudsql-postgres/cloudsql.tf

```diff
@@ -41,11 +41,11 @@ resource "google_sql_database_instance" "cloudsql_instance" {
       }
     }
 
-    #    maintenance_window {
-    #        day             = "${var.cloudsql_maintenance_window_day}"
-    #        hour            = "${var.cloudsql_maintenance_window_hour}"
-    #        update_track    = "${var.cloudsql_maintenance_window_update_track}"
-    #    }
+    maintenance_window {
+      day             = "${var.cloudsql_maintenance_window_day}"
+      hour            = "${var.cloudsql_maintenance_window_hour}"
+      update_track    = "${var.cloudsql_maintenance_window_update_track}"
+    }
```

Comment:
could you use `var.cloudsql_maintenance_window_enable` in conjunction with [conditional attributes](https://stackoverflow.com/questions/51496445/conditional-attributes-in-terraform) here to [omit the args ](https://www.hashicorp.com/blog/terraform-0-12-conditional-operator-improvements#conditionally-omitted-arguments
) if the maintenance window is disabled? Something like:
```
      day             = "${var.cloudsql_maintenance_window_enable == 1 ? var.cloudsql_maintenance_window_day : null}"
      hour            = "${var.cloudsql_maintenance_window_enable == 1 ? var.cloudsql_maintenance_window_hour : null}"
      update_track    = "${var.cloudsql_maintenance_window_enable == 1 ? var.cloudsql_maintenance_window_update_track : null}"
```
note: I don't know if this works, this is hypothetical based on what I read

Reply:
if this works, since `cloudsql_maintenance_window_enable` has a default of 0, it should not install a maintenance window for any other modules that call it, unless those modules actively set `cloudsql_maintenance_window_enable` to 1

Reply:
I tried to dynamically generate the block, does that work?

Reply:
This may be a small nit to pick, but from what I just read about the `dynamic` expression, this may not be the intended use for it. However, if it works, and if we are trying to avoid using CTMPL (or other templating) for conditional rendering, it seems fine.

Contextual note: The entirety of my experience with `dynamic` to date is contained within this comment thread.

Reply:
Yeah definitely a bit of a hack, but a hack that a decent number of people seem to be using since there doesn't seem to be an easier way to do this